### PR TITLE
Fix build errors in xtrabackup-build 8.0.31

### PR DESCRIPTION
**Pull request by gilbertsoft**
Change build script per https://docs.percona.com/percona-xtrabackup/8.0/installation/compiling_xtrabackup.html

### CODE REVIEW

File: .circleci/linux_circle_vm_setup.sh [hunk 1]
```diff
@@ -33,4 +33,8 @@ if ! docker buildx inspect ddev-builder-multi --bootstrap >/dev/null; then docke
 docker buildx inspect --bootstrap
 
 # Install github's gh tool
-wget -O /tmp/gh.deb https://github.com/cli/cli/releases/download/v2.1.0/gh_2.1.0_linux_${ARCH}.deb && sudo dpkg -i /tmp/gh.deb >/dev/null
+curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg \
+&& sudo chmod go+r /usr/share/keyrings/githubcli-archive-keyring.gpg \
+&& echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null \
+&& sudo apt -qq update \
+&& sudo apt -qq install gh -y
```
Comment (rfay) [inline .circleci/linux_circle_vm_setup.sh:40]: I definitely didn't understand why you were installing gh here.

**Review by rfay — APPROVED** [review]
I'm going to go ahead and try this. It still fails with finding zstd on Ubuntu 22.04. 

```
/usr/bin/ld: /tmp/cclmu2LA.ltrans0.ltrans.o: in function `compress_init(char const*) [clone .lto_priv.2]':
/home/ubuntu/workspace/xtrabackup-build/xtrabackup-8.0/percona-xtrabackup/storage/innobase/xtrabackup/src/ds_compress_zstd.cc:62: undefined reference to `ZSTD_createThreadPool'
/usr/bin/ld: /tmp/cclmu2LA.ltrans0.ltrans.o: in function `compress_deinit(ds_ctxt*) [clone .lto_priv.2]':
/home/ubuntu/workspace/xtrabackup-build/xtrabackup-8.0/percona-xtrabackup/storage/innobase/xtrabackup/src/ds_compress_zstd.cc:196: undefined reference to `ZSTD_freeThreadPool'
collect2: error: ld returned 1 exit status
make[2]: *** [storage/innobase/xtrabackup/src/CMakeFiles/xtrabackup.dir/build.make:659: runtime_output_directory/xtrabackup] Error 1
make[1]: *** [CMakeFiles/Makefile2:3698: storage/innobase/xtrabackup/src/CMakeFiles/xtrabackup.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

File: .circleci/linux_circle_vm_setup.sh (hunk 1 above)
Comment (gilbertsoft) [reply .circleci/linux_circle_vm_setup.sh:40]: I changed the installation method to apt only because I always prefer apt over other installation methods.

Comment (rfay) [reply .circleci/linux_circle_vm_setup.sh:40]: The question was why you were installing `gh`, why is it needed?

**Pull request merged**